File: exercises/oop_monkey/lib/monkey.rb
require "pry"
class Monkey
  attr_reader :name, :species, :foods_eaten

  def initialize (name, species)
    @name = name
    @species = species
    @foods_eaten = []
  end

  def eat(food)
    @foods_eaten.push(food)
  end

  def introduce()
    return "Hello, my name is #{@name}, and I'm a #{@species} who has eaten #{@foods_eaten.join(", ")}."
  end
  
end

coco = Monkey.new("coco", "cute monkey")
coco.eat(["bananas", "strabwerries", "fish"])
coco.introduce
binding.pry

File: oop_monkey/lib/monkey.js
// don't forget to use module.exports to export your resulting Monkey constructor!
function Monkey(name, species) {
  this.name = name.charAt(0).toUpperCase() + name.slice(1);
  this.species = species;
  this.foodsEaten = [];
}

Monkey.prototype.eat = function(food) {
  if (!(food.charAt(0).toLowerCase().match(/[aeiou]/i))) {
    this.foodsEaten.push(food)
  }
}

Monkey.prototype.introduce = function() {
  return "Hello, my name is " + this.name + ". I am a " + this.species + ", and I've eaten " + this.foodsEaten
}

module.exports = Monkey
